src/App.tsx

```diff
@@ -1,61 +1,109 @@
+import React, { useState } from 'react';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+import { TodoList } from './components/TodoList';
 
 export const App = () => {
+  const [userSelect, setUserSelect] = useState('0');
+  const [title, setTitile] = useState('');
+  const [todos, setTodos] = useState(todosFromServer);
+  const [submit, setSubmit] = useState(true);
+
+  const trimedTitle = title.trim();
+
+  const largestTodoId = todos.reduce((previously, currently) => (
+    previously.id > currently.id ? previously : currently
+  ));
+
+  const updateTodoList = () => {
+    const newTodo = {
+      id: largestTodoId.id + 1,
+      title: trimedTitle,
+      completed: false,
+      userId: Number(userSelect),
+    };
+
+    setTodos(old => (
+      [...old, newTodo]
+    ));
+    setTitile('');
+    setUserSelect('0');
+    setSubmit(true);
+  };
+
+  const handleSubmit = (event:React.MouseEvent) => {
+    setSubmit(false);
+    event.preventDefault();
+    if (trimedTitle !== '' && userSelect !== '0') {
+      setSubmit(true);
+
+      return updateTodoList();
+    }
+
+    setSubmit(false);
+
+    return null;
+  };
+
   return (
     <div className="App">
       <h1>Add todo form</h1>
 
-      <form action="/api/users" method="POST">
+      <form>
         <div className="field">
-          <input type="text" data-cy="titleInput" />
-          <span className="error">Please enter a title</span>
+          <label htmlFor="title">
+            Title:
+          </label>
+          <input
+            id="title"
+            type="text"
+            data-cy="titleInput"
+            placeholder="Enter a title"
+            value={title}
+            onChange={event => setTitile(event.target.value)}
+          />
+          {
+            !trimedTitle && !submit
+              ? <span className="error">Please enter a title</span>
+              : null
+          }
         </div>
 
         <div className="field">
-          <select data-cy="userSelect">
+          <label htmlFor="user">
+            User:
+          </label>
+          <select
+            id="user"
+            data-cy="userSelect"
+            name="userSelect"
+            value={userSelect}
+            onChange={event => setUserSelect(event.target.value)}
+          >
             <option value="0" disabled>Choose a user</option>
+            {
+              usersFromServer.map(user => (
+                <option value={user.id}>{user.name}</option>
+              ))
+            }
           </select>
-
-          <span className="error">Please choose a user</span>
+          {
+            userSelect === '0' && !submit
+              ? <span className="error">Please choose a user</span>
+              : null
+          }
         </div>
 
-        <button type="submit" data-cy="submitButton">
+        <button
+          type="submit"
+          data-cy="submitButton"
+          onClick={(event) => handleSubmit(event)}
```

Comment:
```suggestion
          onClick={handleSubmit}
```
Process event inside handler function as argument